src/guardian/guardian.service.ts

```diff
@@ -136,14 +137,25 @@ export class GuardianService implements OnModuleInit {
         blockHash: blockData.blockHash,
       });
 
-      await Promise.all(
-        stakingModules.map(async (stakingRouterModule) => {
-          const stakingModuleData =
-            await this.stakingModuleGuardService.getStakingRouterModuleData(
-              stakingRouterModule,
-              blockHash,
-            );
+      // TODO: check only if one of nonce changed
+      const modulesIdWithDuplicateKeys: number[] =
```

Comment:
`handleNewBlock` function is too big.
This function is responsible only to check if are there any updates or not.
Let's move logic below to new function `proceedModules(blockData)`

Reply:
And ping massage generator to `sendPingToBroker(blockData)`